Framework/Core/src/AnalysisDataModel.cxx

```diff
@@ -0,0 +1,113 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/Logger.h"
+#include <boost/algorithm/string.hpp>
+
+namespace o2
+{
+namespace aod
+{
+namespace datamodel
+{
+std::string getTreeName(header::DataHeader dh)
+{
+  std::string description = std::string(dh.dataDescription.str);
+  std::string origin = std::string(dh.dataOrigin.str);
+
+  // lower case of first part of description
+  auto found = description.find_first_of(":");
+  std::string treeName = boost::algorithm::to_lower_copy(description).substr(0, found);
+
+  // add prefix according to origin
+  if (origin == "AOD") {
+    treeName = "O2" + treeName;
+  } else if (origin == "RN2") {
+    treeName = "Run2" + treeName;
+  }
+
+  // exceptions from this
+  if (origin == "AOD" && description == "MCCOLLISLABEL") {
+    treeName = "O2mccollisionlabel";
+  }
+
+  return treeName;
+}
+
+template <typename... C>
+static constexpr auto columnNamesTrait(framework::pack<C...>)
+{
+  return std::vector<std::string>{C::columnLabel()...};
+}
+
+std::vector<std::string> getColumnNames(header::DataHeader dh)
+{
+  std::string description = std::string(dh.dataDescription.str);
+  std::string origin = std::string(dh.dataOrigin.str);
+
+  // get coulmn names
+  // AOD
+  if (origin == "AOD") {
+    if (description == "TRACK:PAR") {
```

Comment:
If you use compile_time_hash for description and hardcoded strings, you can write this as a switch(){} instead of if-else, should be more readable

Reply:
done, although I am not sure if this is more readable?

Reply:
You do not need break after return statements